src/Libraries/Microsoft.Extensions.Telemetry/Sampling/SamplingLoggerBuilderExtensions.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Shared.DiagnosticIds;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.Diagnostics.Sampling;
+
+/// <summary>
+/// Extensions for configuring logging sampling.
+/// </summary>
+[Experimental(diagnosticId: DiagnosticIds.Experiments.Telemetry, UrlFormat = DiagnosticIds.UrlFormat)]
+public static class SamplingLoggerBuilderExtensions
+{
+    /// <summary>
+    /// Adds Trace-based logging sampling to the logging infrastructure. Sampling decisions
+    /// for logs match exactly the sampling decisions for the underlying <see cref="System.Diagnostics.Activity"/>.
+    /// </summary>
+    /// <param name="builder">The dependency injection container to add logging to.</param>
+    /// <returns>The value of <paramref name="builder"/>.</returns>
+    /// <remarks>Please configure Tracing Sampling separately as part of OpenTelemetry .NET.</remarks>
+    /// <exception cref="ArgumentNullException"><paramref name="builder"/> is <see langword="null"/>.</exception>
+    public static ILoggingBuilder AddTraceBasedSampling(this ILoggingBuilder builder)
```

Comment:
```suggestion
    public static ILoggingBuilder AddTraceBasedSampler(this ILoggingBuilder builder)
```
To keep the naming consistent with the other APIs